csharp/examples/IoTExample/Model/SampleDataset.cs

```diff
@@ -0,0 +1,211 @@
+
+                if (success)
+                {
+                    //Console.WriteLine($"Enqueue Task 0");
+                    _size++;
+                }
+                else
+                {
+                    Console.WriteLine("Producing is blocked, percent completed is: {0}%", Math.Round((double)_size / _inputs, 4) * 100);
+                }
+            }
+
+            _rows.CompleteAdding();
+        }
+
+        public async void Consume()
+        {
+            while (!_rows.IsCompleted)
+            {
+                if (!_rows.TryTake(out SensorData item, 3000))
+                {
+                    Console.WriteLine("Consuming is blocked!");
+                }
+                else
+                {
+                    //Console.WriteLine($"Dequeue Task 1");
+                    if (item != null && item.subjectId != null)
+                    {
+                        _colSubjectId.Add((int)item.subjectId);
+                        _colActivityLabel.Add(item.activityLabel);
+                        _colTimestamp.Add((long)item.timestamp);
+                        _colXAxis.Add((double)item.x_Axis);
+                        _colYAxis.Add((double)item.y_Axis);
+                        _colZAxis.Add((double)item.z_Axis);
+                    }
+                }
+
+                if (_colSubjectId.Count == _threshold)
+                {
+                    // Build a record batch using the Fluent API
+                    var recordBatch = new RecordBatch.Builder(_memoryAllocator)
+                        .Append("SubjectId", false, col => col.Int32(array => array.AppendRange(_colSubjectId)))
+                        .Append("ActivityLabel", false, col => col.String(array => array.AppendRange(_colActivityLabel)))
+                        .Append("Timestamp", false, col => col.Int64(array => array.AppendRange(_colTimestamp)))
+                        .Append("XAxis", false, col => col.Double(array => array.AppendRange(_colXAxis)))
+                        .Append("YAxis", false, col => col.Double(array => array.AppendRange(_colYAxis)))
+                        .Append("ZAxis", false, col => col.Double(array => array.AppendRange(_colZAxis)))
+                        .Build();
+
+                    _recordBatches.Add(recordBatch);
+
+                    _colSubjectId.Clear();
+                    _colActivityLabel.Clear();
+                    _colTimestamp.Clear();
+                    _colXAxis.Clear();
+                    _colYAxis.Clear();
+                    _colZAxis.Clear();
+                }
+            }
+        }
+
+        public async Task<bool> PersistData()
+        {
+            string time = DateTime.Now.ToString("yyyyMMdd_HHmmss");
+            Stopwatch stopwatch = new Stopwatch();
+            stopwatch.Start();
+
+            Schema schema = _recordBatches[0].Schema;
+            // Write record batch to a file
+            using (var stream = File.OpenWrite(@"c:\temp\data\iotbigdata.arrow"))
```

Comment:
If this path doesn't exist, you get an error at the end of the run:

```
Producing is blocked, percent completed is: 90%
Unhandled exception. System.IO.DirectoryNotFoundException: Could not find a part of the path 'c:\temp\data\iotbigdata.arrow'.
   at System.IO.FileStream.ValidateFileHandle(SafeFileHandle fileHandle)
   at System.IO.FileStream.CreateFileOpenHandle(FileMode mode, FileShare share, FileOptions options)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
   at System.IO.File.OpenWrite(String path)
   at IoTExample.SampleDataset.PersistData() in C:\git\arrow\csharp\examples\IoTExample\Model\SampleDataset.cs:line 185
   at IoTExample.Program.Main(String[] args) in C:\git\arrow\csharp\examples\IoTExample\Program.cs:line 42
   at IoTExample.Program.<Main>(String[] args)
```

Maybe just write the file to the current directory?

Reply:
Updated.